### PR TITLE
Fixes some flakey tests, add another test

### PR DESCRIPTION
**Pull request by BrandonArp**

### CODE REVIEW

File: src/main/java/com/arpnetworking/metrics/common/tailer/StatefulTailer.java [hunk 1]
```diff
@@ -262,18 +263,7 @@ private void readLoop(final SeekableByteChannel reader) throws IOException, Inte
...
-                        rotate(
-                                Optional.of(reader),
-                                String.format(
-                                        "File rotation detected based last modified time access failure; file=%s",
-                                        _file));
-
-                        // Return to the file loop
-                        return;
-                    }
+                    lastChecked = Optional.of(attributes.getLastModifiedTime());
```
Comment (vjkoskela) [inline src/main/java/com/arpnetworking/metrics/common/tailer/StatefulTailer.java]: How are you guaranteed that this won't fail if the file does not exist? Is the attributes one the new way of doing this?

File: src/test/java/com/arpnetworking/metrics/common/sources/FileSourceTest.java [hunk 2]
```diff
@@ -868,7 +869,11 @@ private void truncate(final Path file) {
         try {
             FileChannel.open(file, StandardOpenOption.WRITE).truncate(0L).close();
         } catch (final IOException e) {
-            // Ignore
+            LOGGER.error()
```
Comment (vjkoskela) [inline src/test/java/com/arpnetworking/metrics/common/sources/FileSourceTest.java]: Should this fail the test?

Comment (BrandonArp) [inline src/test/java/com/arpnetworking/metrics/common/sources/FileSourceTest.java]: Agreed. Lets just let the exception escape and it'll properly fail the test.

File: src/main/java/com/arpnetworking/metrics/common/tailer/StatefulTailer.java (hunk 1 above)
Comment (BrandonArp) [inline src/main/java/com/arpnetworking/metrics/common/tailer/StatefulTailer.java]: Well, lets add a test for it.

Comment (BrandonArp) [inline src/main/java/com/arpnetworking/metrics/common/tailer/StatefulTailer.java]: Actually fixed a test that I believe was intended to test for this case.  The line where attributes is set catches the same exception and does the same rotate.  Everything is good here.

**Comment by BrandonArp** [comment]
Back to you.

**Comment by vjkoskela** [comment]
Looks good to me. Waiting for build to complete.

**Comment by BrandonArp** [comment]
Added another (random) test to bump branch coverage a little bit.

**Pull request merged**